pkg/controller/acmeorders/sync.go

```diff
@@ -347,7 +371,6 @@ func (c *Controller) syncOrderStatus(ctx context.Context, cl acmecl.Interface, o
 
 	acmeOrder, err := cl.GetOrder(ctx, o.Status.URL)
 	if err != nil {
-		// TODO: handle 404 acme responses and mark the order as failed
```

Comment:
This TODO is removed as we now do handle 404 errors in the call-site for syncOrderStatus - if a 404 occurs, we mark the Order as 'Errored' and trigger the back-off logic in the Certificate controller.